Source/Evolve/WarpXEvolve.cpp

```diff
@@ -349,6 +349,11 @@ WarpX::OneStep_nosub (Real cur_time)
                 WarpX::Hybrid_QED_Push(dt);
                 FillBoundaryE(guard_cells.ng_alloc_EB);
             }
+
+            // Synchronize E and B fields on nodal points
+            NodalSyncE();
+            NodalSyncB();
```

Comment:
I've put the synchronization after the Maxwell solve and `FillBoundary` routines and before the `DampPML` routines. Maybe it would be better to put it between the Maxwell solve and `FillBoundary` routines?

Also for now I've only put the synchronization in `onestep_nosub` but maybe we should also add a synchronization in `onestep_sub1`?

Reply:
So I've tested putting the synchronization between the Maxwell solve and `FillBoundary` routines, to see if it changes the benchmarks, as you suggested @EZoni.

It does change a couple of benchmark values but the changes are really small. I've also tested in the instable test-case and this does not affect the instability rate at all. So in the end I don't think the exact position of the synchronization matters much.

Reply:
Thank you for doing these tests, Neil. It's good to know that we can synchronize right before or after `FillBoundary` without much influence on the results.